lib/db/lowlevel.go

```diff
@@ -109,32 +110,45 @@ func (db *Lowlevel) Delete(key []byte, wo *opt.WriteOptions) error {
 }
 
 func (db *Lowlevel) NewIterator(slice *util.Range, ro *opt.ReadOptions) iterator.Iterator {
+	return db.newIterator(func() iterator.Iterator { return db.DB.NewIterator(slice, ro) })
+}
+
+func (db *Lowlevel) newIterator(fn func() iterator.Iterator) iterator.Iterator {
```

Comment:
comment here on why it takes a constructor function (and maybe call it something like that) instead of just an iterator, as it took me a while to figure out what was going on here and why :)